substrate/frame/support/src/traits/tokens/fungible/mod.rs

```diff
@@ -17,26 +17,108 @@
-//!   so should not be called into directly from application code. Other traits depend on this and
-//!   provide default implementations based on it.
-//! - `UnbalancedHold`: Low-level balance mutating functions for balances placed on hold. Does not
+//! Also see the [`tokens_in_substrate`] reference docs for more information about the place of
+//! `fungible` traits in Substrate.
+//!
+//! # Avaliable Traits
+//! - [`Inspect`]: Regular balance inspector functions.
+//! - [`Unbalanced`]: Low-level balance mutating functions. Does not guarantee proper book-keeping
+//!   and so should not be called into directly from application code. Other traits depend on this
+//!   and provide default implementations based on it.
+//! - [`UnbalancedHold`]: Low-level balance mutating functions for balances placed on hold. Does not
 //!   guarantee proper book-keeping and so should not be called into directly from application code.
 //!   Other traits depend on this and provide default implementations based on it.
-//! - `Mutate`: Regular balance mutator functions. Pre-implemented using `Unbalanced`, though the
-//!   `done_*` functions should likely be reimplemented in case you want to do something following
-//!   the operation such as emit events.
-//! - `InspectHold`: Inspector functions for balances on hold.
-//! - `MutateHold`: Mutator functions for balances on hold. Mostly pre-implemented using
-//!   `UnbalancedHold`.
-//! - `InspectFreeze`: Inspector functions for frozen balance.
-//! - `MutateFreeze`: Mutator functions for frozen balance.
-//! - `Balanced`: One-sided mutator functions for regular balances, which return imbalance objects
+//! - [`Mutate`]: Regular balance mutator functions. Pre-implemented using [`Unbalanced`], though
+//!   the `done_*` functions should likely be reimplemented in case you want to do something
+//!   following the operation such as emit events.
+//! - [`InspectHold`]: Inspector functions for balances on hold.
+//! - [`MutateHold`]: Mutator functions for balances on hold. Mostly pre-implemented using
+//!   [`UnbalancedHold`].
+//! - [`InspectFreeze`]: Inspector functions for frozen balance.
+//! - [`MutateFreeze`]: Mutator functions for frozen balance.
+//! - [`Balanced`]: One-sided mutator functions for regular balances, which return imbalance objects
 //!   which guarantee eventual book-keeping. May be useful for some sophisticated operations where
 //!   funds must be removed from an account before it is known precisely what should be done with
 //!   them.
+//!
+//! ## Terminology
+//!
+//! - **Total Issuance**: The total number of units in existence in a system.
+//!
+//! - **Total Balance**: The sum of an account's free and held balances.
+//!
+//! - **Free Balance**: A portion of an account's total balance that is not held.
+//!
+//! - **Held Balance**: Held balance still belongs to the account holder, but is suspended. It can
```

Comment:
nite: there might be some advantage in documenting these items on their corresponding `fn`, as then they will be discoverable when you are for example doing `T::Currency::held_balance(who)`.

Reply:
I hesitate to define the definition of a Hold in every hold `fn`, definition of a Freeze in every freeze `fn`, because it would lead to a lot of comment duplication. 

I'd rather we link back to this ref doc in the doc of each module, which hopefully the dev will look at if they are searching for definitions of holds, freezes, etc.

But I am flexible if you feel strongly about this :)

Reply:
However I do agree it is worth auditing the individual trait method docs and improving those where there is room. Added a TODO for that in the PR checklist.

Reply:
> because it would lead to a lot of comment duplication.

Makes sense :)